### DISCUSSION TITLE
Scalability

### DISCUSSION THREAD

**Discussion by trymeouteh** [mastodon/mastodon#27928] [General]
To my understanding, mastodon cannot scale and this is an issue for mass adoption of mastodon. The scaling issues and concerns are...

- Mastodon instance with many users and usage
- An instance getting request from too many instances and cannot keep up

Will this be address so Mastodon could compete with Twitter? Would it be possible to have an instance with millions of users on it daily and to have instances getting requests from other instances every second and handle the traffic?

**Comment by yamac-kurtulus** [comment]
I am also interested in how much traffic could a small, say a 5$VPS from DigitalOcean could handle (1G Ram, 1 CPU core)

**Comment by sonenko** [comment]
Any numbers or test here?
Because I see that largest instance = 680k users.
But I'm curious how many Users for Mastodon is an Upper limit.
And how many online Users...

**Comment by Gargron** [comment]
There’s no limit known to me if you’re willing to throw more hardware at it. Mastodon scales horizontally, the hardest part is scaling PostgreSQL but that’s not a problem unique to Mastodon and by the point that becomes a bottleneck hopefully you’ll have the means to go further.
